tests/testthat/setup.R

```diff
@@ -1,128 +1,136 @@
-dropAllTablesFromSchema <- function(connectionDetails, schema) {
-  stopifnot(
-    "ConnectionDetails" %in% class(connectionDetails),
-    class(schema) == "character",
-    length(schema) == 1
-  )
-  stopifnot(connectionDetails$dbms %in% c("postgresql", "redshift", "sql server", "oracle"))
-  tableNames <- listTablesInSchema(connectionDetails, schema)
-
-  con <- DatabaseConnector::connect(connectionDetails)
-  on.exit(DatabaseConnector::disconnect(con))
-  dbms <- connectionDetails$dbms
-  if (dbms %in% c("redshift", "postgresql", "sql server")) {
-    for (tableName in tableNames) {
-      DBI::dbExecute(con, paste(
-        "DROP TABLE IF EXISTS",
-        paste(schema, tableName, sep = "."),
-        "CASCADE"
-      ))
-    }
-  } else if (dbms == "oracle") {
-    for (tableName in tableNames) {
-      DBI::dbExecute(con, paste("DROP TABLE IF EXISTS", tableName, "CASCADE"))
-    }
-  }
-}
+testDatabases <- c("postgresql")
+
+if (Sys.getenv("CDM5_POSTGRESQL_SERVER") != "") {
+  library(DatabaseConnector)
+
+  if (dir.exists(Sys.getenv("DATABASECONNECTOR_JAR_FOLDER"))) {
+    jdbcDriverFolder <- Sys.getenv("DATABASECONNECTOR_JAR_FOLDER")
+  } else {
+    jdbcDriverFolder <- file.path(tempdir(), "jdbcDrivers")
+    dir.create(jdbcDriverFolder, showWarnings = FALSE)
+    DatabaseConnector::downloadJdbcDrivers("all", pathToDriver = jdbcDriverFolder)
+
+    Sys.setenv(DATABASECONNECTOR_JAR_FOLDER = jdbcDriverFolder)
+
+    withr::defer({
+      unlink(jdbcDriverFolder,
+             recursive = TRUE,
+             force = TRUE)
+    },
+    testthat::teardown_env())
+  }
+} else {
+  message("Skipping driver setup because environmental variables not set")
+}
+
+# Helper functions used in tests
+getConnectionDetails <- function(dbms) {
+  switch (
+    dbms,
+    "postgresql" = createConnectionDetails(
+      dbms = "postgresql",
+      user = Sys.getenv("CDMDDLBASE_POSTGRESQL_USER"),
+      password = Sys.getenv("CDMDDLBASE_POSTGRESQL_PASSWORD"),
+      server = Sys.getenv("CDMDDLBASE_POSTGRESQL_SERVER"),
+      pathToDriver = jdbcDriverFolder
+    ),
+    "redshift" = createConnectionDetails(
+      dbms = "redshift",
+      user = Sys.getenv("CDMDDLBASE_REDSHIFT_USER"),
+      password = Sys.getenv("CDMDDLBASE_REDSHIFT_PASSWORD"),
+      server = Sys.getenv("CDMDDLBASE_REDSHIFT_SERVER"),
+      pathToDriver = jdbcDriverFolder
+    ),
+    "sql server" = createConnectionDetails(
+      dbms = "sql server",
+      user = Sys.getenv("CDMDDLBASE_SQL_SERVER_USER"),
+      password = Sys.getenv("CDMDDLBASE_SQL_SERVER_PASSWORD"),
+      server = Sys.getenv("CDMDDLBASE_SQL_SERVER_SERVER"),
+      pathToDriver = jdbcDriverFolder
+    ),
+    "oracle" = createConnectionDetails(
+      dbms = "oracle",
+      user = Sys.getenv("CDMDDLBASE_ORACLE_USER"),
+      password = Sys.getenv("CDMDDLBASE_ORACLE_PASSWORD"),
+      server = Sys.getenv("CDMDDLBASE_ORACLE_SERVER"),
+      pathToDriver = jdbcDriverFolder
+    ),
+    "iris" = createConnectionDetails(

```

Comment:
Thanks @bdeboe. Can you confirm that there is an OHDSI iris testing environment?

Reply:
yes, I passed the connection details on to @schuemie a month or so ago for inclusion in the DatabaseConnector test suite.
Do you also want to include IRIS in the CDM regression tests? If so, what kind of schema/database setup would you like (I assume something independent of what the DatabaseConnector tests would see). I can set that up and send you the details through email.

Reply:
We have the following secrets set for testing in the OHDSI GitHub organization:

- CDM_IRIS_CDM_SCHEMA
- CDM_IRIS_CONNECTION_STRING
- CDM_IRIS_OHDSI_SCHEMA
- CDM_IRIS_PASSWORD
- CDM_IRIS_USER

I guess we need an empty schema for testing the DDL? Else we could use CDM_IRIS_OHDSI_SCHEMA

Reply:
Schemas don't need to be created upfront in IRIS, so you can use anything you prefer that wouldn't get in the way of DatabaseConnector's use of the database (which I believe will use the contents of the CDM_IRIS_CDM_SCHEMA env variable / secret). I could also create a separate database (with its own connection string) that'd offer more isolation, but that's probably overkill if we just pick a separate schema name.

Reply:
If we use CDM_IRIS_CDM_SCHEMA that will conflict with the existing tables in that schema which are used for unit testing by other packages, right?

Are you saying we could create a new schema on the fly just for testing the DDL?

Reply:
Thanks! Could you modify the PR to reflect these changes and use the secrets as specified?